### PR TITLE
Fixed washing machine not updating clothes' names if the colour was changed

### LINKED ISSUE #1279: coloring gloves problem
tryed to paint my yellow gloves with mimes grayon, but it didn't work

### PR DESCRIPTION
**Pull request by JohnGinnane**
Most colours will be updated to match the colour of the crayon/stamp
placed in the washing machine. However the descriptions remain unchanged.

Fixes #1279

### CODE REVIEW

**Comment by Nokington** [comment]
k

File: code/game/machinery/washing_machine.dm [hunk 1]
```diff
@@ -65,6 +65,13 @@
 			for(var/obj/item/clothing/I in contents)
 				I.color = wash_color //Simply recolor the items.
 
+				var/colors = list("light brown", "brown", "yellowgreen", "darkred", "lightred", "maroon", "red", "orange", "rainbow", "lightgreen", "green", "lightpurple", "purple", "gold", "darkblue", "lightblue", "aqua", "blue", "yellow", "black", "grey", "gray", "white", "latex", "nitrile", "budget insulated", "insulated", "captain's")
+				
+				for(var/old_color in colors)
+					if(old_color != wash_color)
```
Comment (Carbonhell) [inline code/game/machinery/washing_machine.dm]: I don't get this,i doubt it can work anyhow

Comment (JohnGinnane) [inline code/game/machinery/washing_machine.dm]: I'm looping through a list of common colours used for various clothes like jumpsuits, hats, gloves, shoes, bandanas, etc.

Then I just replace that word with the wash color. On the one hand it an item's name contains more than one colour in it's name they'll both get changed to wash_color, but on the other hand I don't see very many of those items.

I guess the if statement is redundant, but it's basically excluding the potential one case where the old_color matches the wash_color

**Comment by JohnGinnane** [comment]
Here's an example screenshot:

image

**Comment by doobiedoo23** [comment]
So if I get tactical gloves or something and throw them in a washing machine with a crayon, would the name change to '(color of crayon) gloves'?

**Comment by JohnGinnane** [comment]
If the name is "tactical" not yet, but I can add that to the list! Currently discussing other ways to change the name

**Comment by JohnGinnane** [comment]
I've added "combat", and "tactical" to the list of words to update. I've also removed the icon_state being updated because you could end up with results like this (this is actually a bandana):

image

**Comment by Carbonhell** [comment]
Due to it changing capn gloves, tactical/fingerless/etc stuff,it's a feature now so i need a second head ok before merging
code's ok tho

**Comment by doobiedoo23** [comment]
Opens up a few more opportunities to traitors, which is nice. +1

**Comment by Nebyab** [comment]
sure

**Comment by Kokojos** [comment]
two heads :+1: , merging

**Pull request merged**